dataplane/dataplane.cpp

```diff
@@ -1327,6 +1348,28 @@ eResult cDataPlane::splitSharedMemoryPerWorkers()
 
 			ring_id++;
 		}
+		// init lowPriority ring
+		{
+			auto name = "r_lp_" + std::to_string(core_id);
+			auto offset = offsets[shm];
+			auto memaddr = (void*)((intptr_t)shm + offset);
```

Comment:
Avoid c-style casts.
It's generally better to avoid casts whatsoever, but I understand that some things cannot be done without them (though I'm pretty sure we can avoid some of them by using templates and/or `std::variant`, but that's a topic for discussion).

Here we should use named casts: `auto memaddr = reinterpret_cast<void*>(reinterpret_cast<std::uintptr_t>(shm) + offset);`

Reply:
We use C-style typecast in sake of performance and the code is on our most important hot-path.

Also this memory region will contain many data structures (and arrays of them) mapped dynamically with data offsets. Another requirement that the region layout should be parseable by third-party application written in C or Go for example.

However, I think we could improve the approach but let us do this in the another piece of work.

Reply:
Okay, but I cannot resolve the conversation by myself, I guess I don't have the related permissions :)

Reply:
> Unlike static_cast, but like const_cast, the reinterpret_cast expression does not compile to any CPU instructions (except when converting between integers and pointers, or between pointers on obscure architectures where pointer representation depends on its type). It is primarily a compile-time directive which instructs the compiler to treat expression as if it had the type target-type.

So there is no penalty in using `reinterpret_cast`. It also has a benefit of being easily greped